notebooks/12-xarray-advanced.md

```diff
@@ -0,0 +1,598 @@
+    
+- `sf`: snowfall_m
+- `sp`: pressure_pa
+- `t2m`: temperature_k
+- `tp`: precipitation_m
+- `u10`: wind_ms   
+    
+Save the result of the mapping as the variable `era5_renamed`.
+
+<details><summary>Hints</summary>
+
+* Both `rename` and `rename_vars` can be used to rename the DataSet variables
+* The `rename` function requires a `dict-like` input with the current names as the keys and the new names as the values. 
+
+</details>    
+    
+</div>
+
+```{code-cell} ipython3
+:tags: [nbtutor-solution]
+
+mapping = {
+    "sf": "snowfall_m",
+    "sp": "pressure_pa",
+    "t2m": "temperature_k",
+    "tp": "precipitation_m",
+    "u10": "wind_ms"
+}
+
+era5_renamed = era5.rename(mapping)
+era5_renamed
+```
+
+__Note:__ _Make sure you have the variable `era5_renamed` correctly loaded for the following exercises. If not, load the solution of the previous exercise._
+
++++
+
+<div class="alert alert-success">
+
+**EXERCISE**:
+
+Start from the `era5_renamed` variable. You are used to work with temperatures defined in degrees celsius instead of Kelvin. Add a new data variable to `era5_renamed`, named `temperature_c`, by converting the `temperature_k` into degrees celsius:
+    
+$T_{^{\circ}C} = T_{K} - 273.15$
+    
+Create a histogram of the `temperature_c` to check the distribution of all the temperature valus in the data set. Use an appropriate number of bins to draw the histogram.
+
+<details><summary>Hints</summary>
+
+* Xarray - similar to Numpy - applies the mathematical operation element-wise, so no need for loops.
+* Most plot functions work on `DataArray`, so make sure to select the variable to apply the `.plot.hist()`.
+* One can define the number of bins using the `bins` parameter in the `hist` method.
+
+</details>    
+    
+</div>
+
+```{code-cell} ipython3
+:tags: [nbtutor-solution]
+
+era5_renamed["temperature_c"] = era5_renamed["temperature_k"] - 273.15
+```
+
+```{code-cell} ipython3
+:tags: [nbtutor-solution]
+
+era5_renamed["temperature_c"].plot.hist(bins=50);
+```
+
+<div class="alert alert-success">
+
+**EXERCISE**:
+
+Calculate the total snowfall of the entire region of the dataset in function of time and create a line plot showing total snowfall in the y-axis and time in the x-axis.
+
+<details><summary>Hints</summary>
+
+* You need to calculate the total (`sum`) snowfall (`snowfall_m`) in function of time, i.e. aggregate over both the `longitude` and `latitude` dimensions (`dim=["latitude", "longitude"]`).
+* As the result is a DataArray with a single dimension, the default `plot` will show a line, but you can be more explicit by saying `.plot.line()`.
+
+</details>    
+    
+</div>
+
+```{code-cell} ipython3
+:tags: [nbtutor-solution]
+
+era5_renamed["snowfall_m"].sum(dim=["latitude", "longitude"]).plot.line(figsize=(18, 5))
+```
+
+<div class="alert alert-success">
+
```

Comment:
Another idea for a simple exercise here: make a map of the mean temperature (so sum over dim time, and make a map). Currently there is not yet a map in this first set of exercises, and I think that would be nice to add to get a bit an idea of the data (although of course by being a small subset, you don't see much of a contour of belgium in it ..)

Reply:
I'll add just the code that creates a map with cartopy including the belgian country borders. As such, people do have a clue what it the data set about.

Reply:
![image](https://user-images.githubusercontent.com/754862/135731206-6815c129-f308-4969-889f-19259de099f3.png)

Reply:
Ah, perfect, that's a good idea to use cartopy to add some context.

We might have seen cartopy by this point (depending on the order of the other notebooks), in which case this could also be an exercise (to be seen later).